flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/overlays/FlinkDistributionOverlay.java

```diff
@@ -57,9 +59,9 @@
 	final File flinkBinPath;
 	final File flinkConfPath;
 	final File flinkLibPath;
-	final File flinkPluginsPath;
+	final Optional<File> flinkPluginsPath;
 
-	public FlinkDistributionOverlay(File flinkBinPath, File flinkConfPath, File flinkLibPath, File flinkPluginsPath) {
+	public FlinkDistributionOverlay(File flinkBinPath, File flinkConfPath, File flinkLibPath, Optional<File> flinkPluginsPath) {
```

Comment:
https://flink.apache.org/contributing/code-style-and-quality-java.html#java-optional

>    Do not use Optional as a function argument, instead either overload the method or use the Builder pattern for the set of function arguments.
>        Note: an Optional argument can be allowed in a private helper method if you believe that it simplifies the >code (example).
 >   Do not use Optional for class fields.

Reply:
If you insist, then I'll change it to a `@Nullable` parameter. But it won't change much imo.

Reply:
I think that 

> Do not use Optional for class fields.

is a clear statement, and we should respect it because it causes a warning in some setups.

Reply:
Constructor of FlinkDistributionOverlay can be even made private.